DeepSeek V4 Pro + thinking mode: 400 error — reasoning_content not relayed by CLI (v0.15.5)

### What happened?

The bug present in v0.15.4 persists in v0.15.5. When using DeepSeek V4 Pro (DeepSeek API) with thinking mode enabled, the CLI fails to relay reasoning_content tokens back to the API on subsequent requests, causing:

[API Error: 400 The reasoning_content in the thinking mode must be passed back to the API.]

 Session ID: f1f5b1d1-9d99-4e35-b994-67eafa1e178d

    Steps to reproduce:
     1. Configure Qwen Code with DeepSeek API (DeepSeek V4 Pro / reasoning-capable model)
     2. Enable thinking mode
     3. Start any conversation — the error appears on the second or third turn

### Logs

**First occurrence:**

```
      1 2026-04-29T16:20:56.881Z [DEBUG] Received user query (20 chars)
      2 2026-04-29T16:20:58.975Z [ERROR] [OPENAI_ERROR] OpenAI API Error: 400 The `reasoning_content` in the thinking mode must be 
        passed back to the API.
      3 2026-04-29T16:20:58.997Z [ERROR] [ERROR_REPORT] Error when talking to API [Turn.run-sendMessageStream] {
      4   "error": {
      5     "message": "400 The `reasoning_content` in the thinking mode must be passed back to the API.",
      6     "stack": "Error: 400 The `reasoning_content` in the thinking mode must be passed back to the API.
      7     at APIError.generate (file:///.../cli.js:148359:18)
      8     at OpenAI.makeStatusError (file:///.../cli.js:155393:25)
      9     at OpenAI.makeRequest (file:///.../cli.js:155528:29)
     10     ...
     11     at async ContentGenerationPipeline.executeWithErrorHandling (file:///.../cli.js:156899:26)
     12     at async LoggingContentGenerator.generateContentStream (file:///.../cli.js:146500:21)
     13     at async retryWithBackoff (file:///.../cli.js:171879:22)
     14     at async GeminiChat.makeApiCallAndProcessStream (file:///.../cli.js:175197:33)"
     15   }
```


**Second occurrence (same session, ~30s later):**

```
     1 2026-04-29T16:21:14.837Z [ERROR] [OPENAI_ERROR] OpenAI API Error: 400 The `reasoning_content` in the thinking mode must be 
       passed back to the API.
```


### What did you expect to happen?

For the issue to not occur.
Current workaround: Disable thinking mode or switch to a non-reasoning model (e.g. DeepSeek V3).

### Client information


Qwen Code: 0.15.5 (bb32b693a)
Runtime: Node.js v20.20.2 / npm 10.8.2
OS: darwin arm64 (25.3.0)
Auth: API Key - openai
Base URL: https://api.deepseek.com
Model: deepseek-v4-pro
Fast Model: deepseek-v4-pro
Session ID: 19b13bc7-83c6-4f45-b320-72c742943f06
Sandbox: no sandbox
Proxy: no proxy
Memory Usage: 314.0 MB

--- a/packages/core/src/core/openaiContentGenerator/provider/deepseek.test.ts
+++ b/packages/core/src/core/openaiContentGenerator/provider/deepseek.test.ts
@@ -180,9 +180,9 @@ describe('DeepSeekOpenAICompatibleProvider', () => {
     });
 
     // https://github.com/QwenLM/qwen-code/issues/3695 — DeepSeek's thinking
-    // mode rejects subsequent requests when a prior tool-calling assistant
-    // turn omits reasoning_content, even if the model itself returned no
-    // reasoning text. The provider must always send the field.
+    // mode rejects subsequent requests when any prior assistant turn omits
+    // reasoning_content, even if the model itself returned no reasoning text.
+    // The provider must always send the field.
     it('injects empty reasoning_content on tool-calling assistant turns missing it', () => {
       const originalRequest: OpenAI.Chat.ChatCompletionCreateParams = {
         model: 'deepseek-v4-flash',
@@ -245,7 +245,7 @@ describe('DeepSeekOpenAICompatibleProvider', () => {
       expect(assistant.reasoning_content).toBe('Let me glob first.');
     });
 
-    it('does not add reasoning_content to assistant turns without tool_calls', () => {
+    it('injects empty reasoning_content on assistant turns without tool_calls', () => {
       const originalRequest: OpenAI.Chat.ChatCompletionCreateParams = {
         model: 'deepseek-v4-flash',
         messages: [
@@ -259,7 +259,7 @@ describe('DeepSeekOpenAICompatibleProvider', () => {
       const assistant = result.messages?.[1] as {
         reasoning_content?: string;
       };
-      expect(assistant.reasoning_content).toBeUndefined();
+      expect(assistant.reasoning_content).toBe('');
     });
   });
 

--- a/packages/core/src/core/openaiContentGenerator/provider/deepseek.ts
+++ b/packages/core/src/core/openaiContentGenerator/provider/deepseek.ts
@@ -108,17 +108,14 @@ function flattenContentParts(
 }
 
 // DeepSeek's thinking mode requires reasoning_content to be replayed on every
-// prior assistant turn that carried tool_calls. The model may legitimately
-// return a tool round without reasoning text, so the field can be missing
-// when we rebuild the request. Send an empty string in that case so the API
-// contract is satisfied. https://github.com/QwenLM/qwen-code/issues/3695
+// prior assistant turn, including ones without tool_calls. The model may
+// legitimately return a turn without reasoning text, so the field can be
+// missing when we rebuild the request. Send an empty string in that case so
+// the API contract is satisfied. https://github.com/QwenLM/qwen-code/issues/3695
 function ensureReasoningContentOnToolCalls(
   message: OpenAI.Chat.ChatCompletionMessageParam,
 ): OpenAI.Chat.ChatCompletionMessageParam {
   if (message.role !== 'assistant') {
-    return message;
-  }
-  if (!Array.isArray(message.tool_calls) || message.tool_calls.length === 0) {
     return message;
   }
   const extended = message as ExtendedChatCompletionAssistantMessageParam;

